AspNetCore.CacheOutput/InvalidateCacheOutputAttribute.cs

```diff
@@ -26,17 +26,27 @@ public InvalidateCacheOutputAttribute(string methodName, Type controllerType, Ty
             this.cacheKeyGeneratorType = cacheKeyGeneratorType;
         }
 
-        public override async Task OnActionExecutionAsync(ActionExecutingContext context, ActionExecutionDelegate next)
+        private bool IsFailure(int statusCode)
         {
-            await base.OnActionExecutionAsync(context, next);
-
-            if (
-                context.HttpContext.Response != null &&
-                !(
-                    context.HttpContext.Response.StatusCode >= (int)HttpStatusCode.OK && 
-                    context.HttpContext.Response.StatusCode < (int)HttpStatusCode.Ambiguous
-                )
-            )
+            return !(
+                    statusCode >= (int)HttpStatusCode.OK &&
+                    statusCode < (int)HttpStatusCode.Ambiguous
+                );
+        }
+
+        public override async Task OnResultExecutionAsync(ResultExecutingContext context, ResultExecutionDelegate next)
+        {
+            await base.OnResultExecutionAsync(context, next);
```

Comment:
Could we keep old logic and just change OnActionExecutionAsync to OnResultExecutionAsync?

Reply:
By 'old logic' I can only assume you mean the test for Response not being null. I can adjust the code to get the equivelent of that but using the exact same code would be problematic.